### ISSUE TITLE
Make "NA" string configurable

### ISSUE THREAD

**Issue by kerrykimbrough** [Cornutum/tcases#16]
To avoid conflicts in system domains where "NA" has a different meaning.

**Comment by tkruse** [comment]
If you can give a some hint about how you think this should be implemented, I can work on it. I see no straightforward way.

**Comment by tkruse** [comment]
Honestly the clean solution to me seems to be using Java null, and in XML a non value, such as
```
    <Var name="pattern.size" value="ONE"/>
    <Var name="pattern.size"/>  <!-- NA -->
```

**Comment by kerrykimbrough** [comment]
As you've suggested, there's another way to look at this problem. Configuring the NA value is just a hack. The real issue is that the "not-applicable-ness" of a VarValueDef is a distinct property, different from its "name". But Tcases confuses things by encoding them together. A better solution would be to represent these 2 properties distinctly, both in the code for VarValueDef/VarBinding and in the XML representation of a SystemTestDef.

**Comment by kerrykimbrough** [comment]
> the clean solution to me seems to be using Java null, and in XML a non value

I think it's important for the XML to make "not-applicable-ness" explicit and visible. In the code, it could be represented by a null value. But assuming that NA-ness and value are encoded separately, it would be just as well (and more backward-compatible) to keep using "NA".

**Comment by tkruse** [comment]
Actually a test variable value of "null" and "not applicable" might both be different interesting cases. So having a global NA type and allowing null as a value might also be interesting. 

For changes to the XML, you would have to decide how to handle backwards compatibility if you are considering making changes.

A complex approach would be to start versioning the xml schema and supporting multiple versions (e.g. schema1.0 uses `value="NA"`, schema 2.0 uses a new xml tag).

**Comment by kerrykimbrough** [comment]
Fixed by PR #31
